lib/server/socket.js

```diff
@@ -8,7 +8,13 @@ module.exports = ({ server, filters, config }) => {
   const io = new Primus(server, {
     transformer: 'engine.io',
     parser: 'EJSON',
-    maxLength: '20971520',
+    // engine.io < 4.0.0 is vulnerable to Denial of Service attack. To mitigate
+    // the vulnerability we reduce HTTP buffer size down to 1mb as it is done
+    // in the official fix commit.
+    // For more details:
+    // - https://app.snyk.io/vuln/SNYK-JS-ENGINEIO-1056749
+    // - https://github.com/socketio/engine.io/commit/734f9d1268840722c41219e69eb58318e0b2ac6b
+    maxLength: 1e6,
```

Comment:
This has an impact on Broker communications, we raised it to 20MB a few weeks ago because payloads bigger than 10MB were being blocked. This is going to make things a lot worse

Reply:
Would you mind to add a comment about it in the code?

Reply:
Adding comment: https://github.com/snyk/broker/pull/341